pkg/deploy/upgrade.go

```diff
@@ -102,3 +114,39 @@ func (d *deployer) removeOldScaleset(ctx context.Context, vmssName string) error
 	d.log.Printf("deleting scaleset %s", vmssName)
 	return d.vmss.DeleteAndWait(ctx, d.config.ResourceGroupName, vmssName)
 }
+
+// saveRPVersion for current location in shared storage account for environment
+func (d *deployer) saveRPVersion() error {
+	t := time.Now().UTC().Truncate(time.Second)
+	res, err := d.accounts.ListAccountSAS(
+		context.Background(), *d.config.Configuration.GlobalResourceGroupName, *d.config.Configuration.RPVersionStorageAccountName, mgmtstorage.AccountSasParameters{
+			Services:               mgmtstorage.B,
+			ResourceTypes:          mgmtstorage.SignedResourceTypesO,
+			Permissions:            "cw", // create and write
+			Protocols:              mgmtstorage.HTTPS,
+			SharedAccessStartTime:  &date.Time{Time: t},
+			SharedAccessExpiryTime: &date.Time{Time: t.Add(24 * time.Hour)},
+		})
+	if err != nil {
+		return err
+	}
+
+	v, err := url.ParseQuery(*res.AccountSasToken)
+	if err != nil {
+		return err
+	}
+
+	// one container per versioned entity (e.g. RP)
+	blobClient := azstorage.NewAccountSASClient(
+		*d.config.Configuration.RPVersionStorageAccountName, v, azure.PublicCloud).GetBlobService()
+
+	containerRef := blobClient.GetContainerReference("rpVersion")
+	_, err = containerRef.CreateIfNotExists(nil)
+	if err != nil {
+		return err
+	}
```

Comment:
I suspect your SAS token doesn't have rights to do this.  The way this works in the RP cluster creation code is that the container is deployed as part of the ARM template.  Unless this definitely works, probably best to do the same here.

Reply:
Done